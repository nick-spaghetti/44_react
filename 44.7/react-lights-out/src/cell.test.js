import React from "react";
import Cell from "./Cell";
import { render, fireEvent } from "@testing-library/react";

describe("<Cell /> rendering", () => {
	let container;
	beforeEach(() => {
		let tr = document.createElement("tr");
		container = document.body.appendChild(tr);
	});

	it("should render without breaking", () => {
		render(<Cell />, { container });
	});

	it("matches snapshot", () => {
		const { asFragment } = render(<Cell isLit />, { container });
		expect(asFragment()).toMatchSnapshot();
	});

	it("matches snapshot when not lit", () => {
		const { asFragment } = render(<Cell />, { container });
		expect(asFragment()).toMatchSnapshot();
	});
});
