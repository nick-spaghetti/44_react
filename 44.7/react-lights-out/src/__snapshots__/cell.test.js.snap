// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell /> rendering matches snapshot 1`] = `<DocumentFragment />`;

exports[`<Cell /> rendering matches snapshot when not lit 1`] = `<DocumentFragment />`;
